### PR TITLE
feat: add fetch server config api and usecase [AR-1037]

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

### Issues
[AR-1037]

### Solutions
Add api, remoteRepository, repository and useCase for fetch server config.

### Testing

WIP

### TODO

Refactor naming of the models

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[AR-1037]: https://wearezeta.atlassian.net/browse/AR-1037

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class ServerConfigApiImp(private val httpClient: HttpClient) : ServerConfigApi {
+
+    /**
+     * Fetch remote configuration
+     * @param serverConfigUrl the remote config url
+     */
+    override suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<NetworkConfigDTO> = wrapKaliumResponse {
+        httpClient.get(serverConfigUrl)
+    }
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt:20]: wondring if it's a good idea to rename `NetworkConfigDTO` to something like `private data class FetchServerConfigResponse`
```suggestion
    override suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<BackendConfig> = wrapKaliumResponse<NetworkConfigDTO> {
        httpClient.get(serverConfigUrl)
    }.mapSuccess { // map the response to BackendConfig }
```
or we can delete `BackendConfig` class and change `AuthenticatedNetworkContainer` to pass `NetworkConfigDTO`
https://github.com/wireapp/kalium/blob/feat/AR-1037_fetch_server_config/network/src/commonMain/kotlin/com/wire/kalium/network/AuthenticatedNetworkContainer.kt#L35
just so we don't have 2 models to the same thing inside `network`

**Review by MohamadJaara — COMMENTED** [review]
great work, only one suggestion about `network` exposing 2 models for the same thing

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt (hunk 1 above)
Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt]: Thanks @MohamadJaara , I good suggestion. Did it as you suggested.

**Review by MohamadJaara — APPROVED** [review]
top effort 🚀

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Fetch remote configuration
+     * @param serverConfigUrl the remote config url
+     */
+    // TODO: use wrapApiRequest once PR #226 is merged
+    override suspend fun fetchServerConfig(serverConfigUrl: String): NetworkResponse<BackendConfig> =
+        wrapKaliumResponse<ServerConfigResponse> {
+            httpClient.get(serverConfigUrl)
+        }.mapSuccess {
+            print("xxxxxx ${it.toString()}")
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt:25]: 👀 logging is coming soon

Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/configuration/ServerConfigApi.kt:25]: Oh :D forgot to remove that :)) I'll do that.

**Pull request merged**